### PR TITLE
Fix instagram image display

### PR DESCRIPTION
**Pull request by michaelha**

### CODE REVIEW

**Comment by michaelha** [comment]
@richardtape Please review this as well as #2   Feel free to email/slack dm me to discuss if you have any feedback.  Thanks!

**Comment by richardtape** [comment]
Looks like you have pushed both commits to this pull request so now there's conflicts. If you could please revise and resolve them, I'll take a look :)

**Comment by michaelha** [comment]
Will do it right now :)

**Comment by michaelha** [comment]
@richardtape ready for review.

File: foa-site.php [hunk 1]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
 	function scrape_instagram( $username, $slice = 9 ) {
 		$username = strtolower( $username );
 		if ( false === ( $instagram = get_transient( 'instagram-media-news-'.sanitize_title_with_dashes( $username ) ) ) ) {
-			$remote = wp_remote_get( 'http://instagram.com/'.trim( $username ) );
+			$remote = wp_remote_get( 'https://instagram.com/'.trim( $username ) );
```
Comment (michaelha) [inline foa-site.php]: 1304 | WARNING | [ ] wp_remote_get() is highly discouraged, please use vip_safe_wp_remote_get() instead.

File: foa-site.php [hunk 2]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
 			if ( is_wp_error( $remote ) )
 				throw new Exception('Unable to communicate with Instagram.');
 			if ( 200 != wp_remote_retrieve_response_code( $remote ) )
-				return 'Instagram did not return a 200.';
+				throw new Exception('Instagram did not return a 200.');
```
Comment (michaelha) [inline foa-site.php]: 1308 | ERROR   | [x] Expected 1 spaces after opening bracket; 0 found
 1308 | ERROR   | [x] Expected 1 spaces before closing bracket; 0 found

File: foa-site.php [hunk 3]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
 				throw new Exception('Instagram has returned invalid data.');
-			// old style
-			if ( isset( $insta_array['entry_data']['UserProfile'][0]['userMedia'] ) ) {
-				$images = $insta_array['entry_data']['UserProfile'][0]['userMedia'];
-				$type = 'old';
-			// new style
-			} else if ( isset( $insta_array['entry_data']['ProfilePage'][0]['user']['media']['nodes'] ) ) {
-				$images = $insta_array['entry_data']['ProfilePage'][0]['user']['media']['nodes'];
-				$type = 'new';
+			if ( isset( $insta_array['entry_data']['ProfilePage'][0]['graphql']['user']['edge_owner_to_timeline_media']['edges'] ) ) {
```
Comment (michaelha) [inline foa-site.php]: no error.

File: foa-site.php [hunk 4]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+				$images = $insta_array['entry_data']['ProfilePage'][0]['graphql']['user']['edge_owner_to_timeline_media']['edges'];
```
Comment (michaelha) [inline foa-site.php]: no error.

File: foa-site.php [hunk 5]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
-							'time'		  	=> $image['date'],
-							'comments'	  	=> $image['comments']['count'],
-							'likes'		 	=> $image['likes']['count'],
-							'thumbnail'	 	=> $image['display_src'],
-							'type'		  	=> $type
-						);
-					} 
-				break;
-			}
+            foreach ( $images as $image ) {
```
Comment (michaelha) [inline foa-site.php]: 1322 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

File: foa-site.php [hunk 6]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                $image['display_src'] = preg_replace( "/^http:/i", "", $image['node']['display_url'] );
```
Comment (michaelha) [inline foa-site.php]: 1323 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1323 | ERROR   | [x] String "/^http:/i" does not require double quotes; use single quotes instead
 1323 | ERROR   | [x] String "" does not require double quotes; use single quotes instead

File: foa-site.php [hunk 7]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                if ( $image['is_video']  == true ) {
```
Comment (michaelha) [inline foa-site.php]: 1324 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1324 | WARNING | [ ] Found: ==. Use strict comparisons (=== or !==).
 1324 | ERROR   | [ ] Use Yoda Condition checks, you must.
 1324 | ERROR   | [x] Expected 1 space before "=="; 2 found

File: foa-site.php [hunk 8]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                    $type = 'video';
```
Comment (michaelha) [inline foa-site.php]: 1325 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

File: foa-site.php [hunk 9]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                } else {
```
Comment (michaelha) [inline foa-site.php]: 1326 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

File: foa-site.php [hunk 10]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                    $type = 'image';
```
Comment (michaelha) [inline foa-site.php]: 1327 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

File: foa-site.php [hunk 11]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                }
```
Comment (michaelha) [inline foa-site.php]: 1328 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1329 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

File: foa-site.php [hunk 12]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                $instagram[] = array(
+                    'description'   => __( $image['node']['edge_media_to_caption']['edges'][0]['node']['text'], 'wpiw' ),
```
Comment (michaelha) [inline foa-site.php]: 1330 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1330 | WARNING | [x] Array double arrow not aligned correctly; expected 1 space(s) between "'description'" and double arrow, but found 3.
 1330 | ERROR   | [ ] The $text arg must be a single string literal, not "$image['node']['edge_media_to_caption']['edges'][0]['node']['text']".

File: foa-site.php [hunk 13]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                    'link'		  	=> '//instagram.com/p/' . $image['node']['shortcode'],
```
Comment (michaelha) [inline foa-site.php]: 1331 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1331 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1331 | WARNING | [x] Array double arrow not aligned correctly; expected 8 space(s) between "'link'" and double arrow, but found 10.

File: foa-site.php [hunk 14]
```diff
@@ -1301,72 +1301,41 @@ function instagram_items_to_html($instagram_items){
...
+                    'time'		  	=> $image['node']['taken_at_timestamp'],
```
Comment (michaelha) [inline foa-site.php]: 1332 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1332 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1332 | WARNING | [x] Array double arrow not aligned correctly; expected 8 space(s) between "'time'" and double arrow, but found 10.
 1333 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1333 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1333 | WARNING | [x] Array double arrow not aligned correctly; expected 4 space(s) between "'comments'" and double arrow, but found 6.
 1334 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1334 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1334 | WARNING | [x] Array double arrow not aligned correctly; expected 7 space(s) between "'likes'" and double arrow, but found 9.
 1335 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1335 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1335 | WARNING | [x] Array double arrow not aligned correctly; expected 3 space(s) between "'thumbnail'" and double arrow, but found 5.
 1336 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1336 | ERROR   | [x] Spaces must be used for mid-line alignment; tabs are not allowed
 1336 | WARNING | [x] Array double arrow not aligned correctly; expected 8 space(s) between "'type'" and double arrow, but found 10.
 1336 | ERROR   | [x] Each array item in a multi-line array declaration must end in a comma
 1337 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed
 1338 | ERROR   | [x] Tabs must be used to indent lines; spaces are not allowed

**Review by michaelha — COMMENTED** [review]
fix phpcs errors

File: foa-site.php (hunk 1 above)
Comment (michaelha) [reply foa-site.php]: ignore.

**Comment by michaelha** [comment]
@richardtape I've addressed phpcs error messages. Please review this PR.  Thanks.

File: foa-site.php [hunk 15]
```diff
@@ -1301,71 +1301,40 @@ function instagram_items_to_html($instagram_items){
 	function scrape_instagram( $username, $slice = 9 ) {
 		$username = strtolower( $username );
 		if ( false === ( $instagram = get_transient( 'instagram-media-news-'.sanitize_title_with_dashes( $username ) ) ) ) {
-			$remote = wp_remote_get( 'http://instagram.com/'.trim( $username ) );
+			$remote = wp_remote_get( 'https://instagram.com/'.trim( $username ) );
 			if ( is_wp_error( $remote ) )
-				throw new Exception('Unable to communicate with Instagram.');
+				throw new Exception( 'Unable to communicate with Instagram.' );
```
Comment (richardtape) [inline foa-site.php]: Conditionals should always have braces. i.e.

```
if ( is_wp_error( $remote ) ) {
    throw new Exception( 'Unable to communicate with Instagram.' );
}
```

This applies to a few chunks in this commit.

File: foa-site.php [hunk 16]
```diff
@@ -1301,71 +1301,40 @@ function instagram_items_to_html($instagram_items){
...
 			if ( 200 != wp_remote_retrieve_response_code( $remote ) )
```
Comment (richardtape) [inline foa-site.php]: I think `!=` should be `!==` to have strict adherence to coding standards

File: foa-site.php [hunk 17]
```diff
@@ -1301,71 +1301,40 @@ function instagram_items_to_html($instagram_items){
 	function scrape_instagram( $username, $slice = 9 ) {
 		$username = strtolower( $username );
 		if ( false === ( $instagram = get_transient( 'instagram-media-news-'.sanitize_title_with_dashes( $username ) ) ) ) {
-			$remote = wp_remote_get( 'http://instagram.com/'.trim( $username ) );
+			$remote = wp_remote_get( 'https://instagram.com/'.trim( $username ) );
```
Comment (richardtape) [inline foa-site.php]: Should be spaces either side of the concatenating period. i.e. 

```
$remote = wp_remote_get( 'https://instagram.com/' . trim( $username ) );
```

**Review by richardtape — CHANGES_REQUESTED** [review]
This is nearly ready to go. Just a couple tweaks.

File: foa-site.php (hunk 17 above)
Comment (richardtape) [reply foa-site.php]: Actually, whilst I'm thinking about it, please also add an `esc_url()` check around the whole URL.

i.e.

```
$remote = wp_remote_get( esc_url( 'https://instagram.com/' . trim( $username ) ) );
```

File: foa-site.php [hunk 18]
```diff
@@ -1301,23 +1301,27 @@ function instagram_items_to_html($instagram_items){
 	function scrape_instagram( $username, $slice = 9 ) {
 		$username = strtolower( $username );
 		if ( false === ( $instagram = get_transient( 'instagram-media-news-'.sanitize_title_with_dashes( $username ) ) ) ) {
-			$remote = wp_remote_get( 'https://instagram.com/'.trim( $username ) );
-			if ( is_wp_error( $remote ) )
+			$remote = wp_remote_get( 'https://instagram.com/' . trim( $username ) );
```
Comment (richardtape) [inline foa-site.php]: This is my bad, see my other comment on this. I think we should have an `esc_url()` check on this whole URL.

**Review by richardtape — CHANGES_REQUESTED** [review]

**Review by richardtape — APPROVED** [review]
👍

**Pull request merged**